src/encode.c

```diff
@@ -845,29 +851,41 @@ PyImaging_LibTiffEncoderNew(PyObject* self, PyObject* args)
 
     for (pos = 0; pos < d_size; pos++) {
         key = PyList_GetItem(keys, pos);
+        key_int = (int)PyInt_AsLong(key);
         value = PyList_GetItem(values, pos);
         status = 0;
-        TRACE(("Attempting to set key: %d\n", (int)PyInt_AsLong(key)));
+        core = 0;
+        for (i=0; i<32; i++) {
```

Comment:
Maybe remove the hardcoded array size?
```suggestion
        int number_of_tags = sizeof(tags) / sizeof(int);
        for (i=0; i<number_of_tags; i++) {
```